lib/tagging_migrator.rb

```diff
@@ -0,0 +1,58 @@
+# TaggingMigrator
+#
+# We are migrating the taggings from panopticon/content-api to publishing-api.
+# This class takes the taggings for a certain app and sends it to the
+# publishing-api so that it contains the current taggings. This functionality
+# will be removed after the migration.
+
+require 'gds_api/publishing_api_v2'
+
+class TaggingMigrator
+  def initialize(app_name)
+    @app_name = app_name
+  end
+
+  def migrate!
+    artefacts_owned_by_app.each do |artefact|
+      next unless artefact.content_id
+      migrate_tags_for_artefact(artefact)
+    end
+  end
+
+private
+
+  def migrate_tags_for_artefact(artefact)
+    link_payload = {
+      mainstream_browse_pages: [],
+      topics: [],
+      organisations: [],
+    }
+
+    artefact.tags.each do |tag|
+      if tag.tag_type == 'section'
+        link_payload[:mainstream_browse_pages] << tag.content_id
+      end
+
+      if tag.tag_type == 'specialist_sector'
+        link_payload[:topics] << tag.content_id
+      end
+
+      if tag.tag_type == 'organisation'
+        link_payload[:organisations] << tag.content_id
+      end
+    end
```

Comment:
Answer - no, because related links aren't tags.  We'll deal with them when we get to the relevant publishing apps.